### PR TITLE
add wrapped bridge token

### PR DESCRIPTION
**Pull request by xiaozhou**

### CODE REVIEW

**Comment by Goodman2025** [comment]
@freedom for all people in this world.

**Review by liuxiaobleach — APPROVED** [review]

File: contracts/pegged-bridge/tokens/WrappedBridgeToken.sol [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    function mint(address _to, uint256 _amount) external onlyBridge returns (bool) {
+        _mint(address(this), _amount);
+        IERC20(canonical).safeTransfer(_to, _amount);
+        return true;
+    }
+
+    function burn(address _from, uint256 _amount) external onlyBridge returns (bool) {
+        _burn(address(this), _amount);
```
Comment (Dominator008) [inline contracts/pegged-bridge/tokens/WrappedBridgeToken.sol:47]: Add
```
_spendAllowance(_from, msg.sender, _amount);
```
?

Comment (xiaozhou) [reply contracts/pegged-bridge/tokens/WrappedBridgeToken.sol:47]: the `safeTransferFrom` at line 48 already requires user approval of the canonical token to this contract.

File: contracts/pegged-bridge/tokens/WrappedBridgeToken.sol [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        IERC20(canonical).safeTransferFrom(_from, address(this), _amount);
+        return true;
+    }
+
+    function addLiquidity(uint256 _amount) external {
+        IERC20(canonical).safeTransferFrom(msg.sender, address(this), _amount);
```
[the remark this answers is not in the document]
Comment (xiaozhou) [reply contracts/pegged-bridge/tokens/WrappedBridgeToken.sol]: done

**Pull request merged**